File: STRIVER/video/vid-7/rev-number.cpp
#include <bits/stdc++.h>
using namespace std;

// time complexity  = O(log(x))

int rev_number(int N)
{
    int n = N;
    long long int rev_num = 0;

    if (N < 0)
        n = -n;

    while (n != 0)
    {
        int temp = n % 10;
        rev_num = (rev_num * 10) + temp;
        n = n / 10;
    }

    if (rev_num > INT_MAX || rev_num < INT_MIN)
        return 0;

    if (N < 0)
        return -rev_num;
    else
        return rev_num;
}

int main()
{
    int N = 123;
    cout << rev_number(N) << endl;
    return 0;
}

File: STRIVER/video/vid-7 (Maths)/rev-number.py
import sys
#  time complexity  = O(Log(X))

MIN_SIZE = -2**31 
MAX_SIZE = 2**31 - 1

def evenlyDivides(N):
    # code here
    n = N
    rev_num = 0
    
    
    if(str(N)[0] == '-'):
        n = int(str(N)[1:])
    while n!=0 :
        temp = n%10
        rev_num = (rev_num * 10) + temp
        n = n // 10

    if(rev_num > MAX_SIZE or rev_num < MIN_SIZE):
            return 0
    
    if(str(N)[0] == '-'):
        return int('-'+str(rev_num))
    else:
        return rev_num

print(evenlyDivides(1534236469))